libs/knowledge-store/ragstack_knowledge_store/graph_store.py

```diff
@@ -215,47 +208,33 @@ def __init__(
             LIMIT ?
             """
         )
-        self._query_ids_and_embedding_by_embedding.consistency_level = (
-            ConsistencyLevel.QUORUM
-        )
-
-        self._query_linked_ids = session.prepare(
-            f"""
-            SELECT target_content_id AS content_id
-            FROM {keyspace}.{edge_table}
-            WHERE source_content_id = ?
-            """
-        )
+        self._query_ids_and_embedding_by_embedding.consistency_level = ConsistencyLevel.ONE
 
-        self._query_edges_by_source = session.prepare(
+        self._query_source_tags_by_ids = session.prepare(
             f"""
-            SELECT target_content_id, target_text_embedding
-            FROM {keyspace}.{edge_table}
-            WHERE source_content_id = ?
-            """
-        )
-
-        self._query_ids_by_link_to_tag = session.prepare(
-            f"""
-            SELECT content_id
+            SELECT link_to_tags
             FROM {keyspace}.{node_table}
-            WHERE link_to_tags CONTAINS ?
+            WHERE content_id IN ?
             """
         )
 
-        self._query_ids_and_embedding_by_link_to_tag = session.prepare(
+        # TODO: These queries require `ALLOW FILTERING` when run against Cassandra docker images.
```

Comment:
Using ALLOW FILTERING is extremely slow.
Can we do it with maybe partitionning ?

Reply:
I think the ALLOW FILTERING isn't needed in Astra, but is in Cassandra. I can double check. I think the alternative is to blast out a bunch of separate queries rather than using the `IN`, which may actually be *faster*.

But, may be best to make that change as a follow-up so that we can get the bulk of the changes (including API changes) in. This would be a change under the hood, so to speak.

Reply:
Done.